internal/mode/static/telemetry/collector.go

```diff
@@ -275,3 +259,59 @@ func CollectClusterID(ctx context.Context, k8sClient client.Reader) (string, err
 	}
 	return string(kubeNamespace.GetUID()), nil
 }
+
+// CollectNodeCount returns the number of nodes in the cluster.
+func CollectNodeCount(ctx context.Context, k8sClient client.Reader) (int, error) {
+	var nodes v1.NodeList
+	if err := k8sClient.List(ctx, &nodes); err != nil {
+		return 0, fmt.Errorf("failed to get NodeList: %w", err)
+	}
+
+	return len(nodes.Items), nil
+}
+
+type clusterInformation struct {
+	Platform  string
+	Version   string
+	ClusterID string
+	Nodes     v1.NodeList
+}
+
+func collectClusterInformation(ctx context.Context, k8sClient client.Reader) (clusterInformation, error) {
+	var clusterInfo clusterInformation
+	var err error
+
+	var nodes v1.NodeList
+	if err = k8sClient.List(ctx, &nodes); err != nil {
+		return clusterInformation{}, fmt.Errorf("failed to get NodeList: %w", err)
+	}
+	if len(nodes.Items) == 0 {
+		return clusterInformation{}, errors.New("failed to collect cluster information: NodeList length is zero")
+	}
+
+	clusterInfo.Nodes = nodes
+
+	var clusterID string
+	if clusterID, err = CollectClusterID(ctx, k8sClient); err != nil {
+		return clusterInformation{}, fmt.Errorf("failed to collect cluster information: %w", err)
+	}
+	clusterInfo.ClusterID = clusterID
+
+	node := nodes.Items[0]
+
+	clusterInfo.Version = "unknown"
+	kubeletVersion := node.Status.NodeInfo.KubeletVersion
+
+	if version, err := parseSemver(kubeletVersion); err == nil {
```

Comment:
Could we utilize this function https://pkg.go.dev/k8s.io/apimachinery/pkg/util/version#ParseGeneric ?

Reply:
Unfortunately not : ( 


`"v1.27.8-gke.1067004"` is a valid kubectl version.

However `ParseGeneric` has this in its documentation:

> The version string must consist of two or more dot-separated numeric fields

So the `8-gke` portion would cause the function to error. 

And in the code it has it here

```
num, err := strconv.ParseUint(comp, 10, 0)
		if err != nil {
			return nil, fmt.Errorf("illegal non-numeric version component %q in %q: %v", comp, str, err)
		}

```

Reply:
```go
	version, err := version.ParseGeneric("v1.27.8-gke.1067004")
	fmt.Println(version, err)
```

Output:
```
1.27.8 <nil>
```

Reply:
I misunderstood the code on `version.ParseGeneric`. I spent some time on the regex expression it uses and also tested each of the kubeletversions we get from each of the providers and it looks like we will be able to use this! 

Thanks a lot @pleshakov this will take the burden off of needing to write the perfect function for parsing the semver string and needing to test it as we will most likely be able to rely on this `apimachinery` exported function! Thanks for catching that!

Reply:
It would be unnecessary to test their code through various kubelet versions right?